dms/models/ir_attachment.py

```diff
@@ -0,0 +1,64 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _get_dms_directories(self, res_model, res_id):
+        return self.env["dms.directory"].search(
+            [("res_model", "=", res_model), ("res_id", "=", res_id)]
+        )
+
+    def _dms_operations(self, attachment_ids):
+        for attachment in attachment_ids:
+            if attachment.res_model and attachment.res_id:
+                directories = self._get_dms_directories(
+                    attachment.res_model, attachment.res_id
+                )
+                if not directories:
+                    # Auto-create directories
+                    directories_without_res_id = self._get_dms_directories(
+                        attachment.res_model, False
+                    )
+                    for directory in directories_without_res_id:
+                        model_item = (
+                            self.env[attachment.res_model]
+                            .sudo()
+                            .browse(attachment.res_id)
+                        )
+                        self.env["dms.directory"].create(
+                            {
+                                "name": model_item.display_name,
+                                "res_model": attachment.res_model,
+                                "res_id": attachment.res_id,
+                                "parent_id": directory.id,
+                                "storage_id": directory.storage_id.id,
+                            }
+                        )
+                    # Get dms_directories again (with items previously created)
+                    directories = self._get_dms_directories(
+                        attachment.res_model, attachment.res_id
+                    )
+                # Auto-create_files
+                if directories:
```

Comment:
No need of this `if`. If no `directories`, `for` loop doesn't iterate. If the problem is that `directories` can be `False` or `None`, then please change that method for always returning a recordset, but empty. It's a good practice to be homogeneous on the returned value for a method.